tools/c7n_azure/tests/test_session.py

```diff
@@ -64,14 +64,17 @@ def test_initialize_session_auth_file_custom_subscription(self):
 
             self.assertIs(type(s.get_credentials()), ServicePrincipalCredentials)
             self.assertEqual(s.get_subscription_id(), CUSTOM_SUBSCRIPTION_ID)
-            self.assertEqual(s.get_tenant_id(), 'tenant')
+
+            # will vary between recorded/live auth options but useful to ensure
+            # we ended up with one of the valid values
+            self.assertTrue(s.get_tenant_id() in [DEFAULT_TENANT_ID, 'tenant'])
 
     def test_initialize_session_principal(self):
         with patch('azure.common.credentials.ServicePrincipalCredentials.__init__',
                    autospec=True, return_value=None):
             with patch.dict(os.environ,
                             {
-                                constants.ENV_TENANT_ID: 'tenant',
+                                constants.ENV_TENANT_ID: DEFAULT_TENANT_ID,
                                 constants.ENV_SUB_ID: DEFAULT_SUBSCRIPTION_ID,
```

Comment:
self.session.get_subscription_id?

Reply:
I don't follow, I think this is correct.  It is just testing that whatever we pass in here is what we get out.